### ISSUE TITLE
Synching colorranges on the same axis?

### ISSUE THREAD

**Issue by piever** [JuliaPlots/Makie.jl#912]
I've just noticed that it is a bit problematic to use a numeric color attributes in separate plots in the same axis.

For example, the following:

```julia
using CairoMakie

f = Figure()
ax = Axis(f[1, 1])
x1, y1, z1 = 1:10, 1:10, rand(10)
x2, y2, z2 = 11:20, 1:10, rand(10) .+ 100 
scatter!(ax, x1, y1, color = z1)
scatter!(ax, x2, y2, color = z2)
f
```

generates this figure

![test](https://user-images.githubusercontent.com/6333339/114587799-46a98e80-9c86-11eb-8d8e-5b5eee9a8a00.png)

whereas IMO it should look something like this:

![test2](https://user-images.githubusercontent.com/6333339/114587847-54f7aa80-9c86-11eb-99e5-2e5aeed9146b.png)

That is to say, I imagine that the "colorrange" transform should be a property of the axis, pretty much like the x and y limits.

I'm not sure what is the best fix is in practice, maybe there could be some `linkcolors!` analogous to `linkaxes!`?

**Comment by jkrumbiegel** [comment]
That sure is confusing, although I wouldn't want to restrict one axis to one colormap. People want to do weird things with multiple scales sometimes. `linkcolors` would then have to work on the plot level, splicing the same observable in as the colorrange maybe?

**Comment by piever** [comment]
> That sure is confusing, although I wouldn't want to restrict one axis to one colormap. People want to do weird things with multiple scales sometimes.

Good point. I personally think that that's a bit of a questionable practice, but it should be up to the user to decide.

> `linkcolors` would then have to work on the plot level, splicing the same observable in as the colorrange maybe?

Yes, I imagine `linkcolors` would take either a collection of plots (say `linkcolors(plots::AbstractPlot...)`) or a collection of axes (say `linkcolors(axs::Axis...)`), in which case, it would consider all plots with a colorrange within those axes. Then it would just sync all colorranges to be the "shared range".

I'm not sure if it should actually replace the observables to be the same observable, or sync those that exist already. I'm not completely up to speed with the latest versions of Observables so I'm not sure how easy it is to sync a bunch of observables without triggering a large amount of callbacks all the time.
